### ISSUE TITLE
Increase the code coverage of environ-linux.cpp

### ISSUE THREAD

**Issue by alabulei1** [WasmEdge/WasmEdge#1671]
## Motivation

According to the codecov report, we should increase [environ-linux.cpp](https://github.com/WasmEdge/WasmEdge/blob/master/lib/host/wasi/environ-linux.cpp)

## Appendix

[Codecov report for inode-linux.cpp](https://app.codecov.io/gh/WasmEdge/WasmEdge/blob/master/lib/host/wasi/environ-linux.cpp)

**Comment by Sush-Dew** [comment]
@alabulei1 Could you assign this issue to me? I am a beginner and this issue seems a good first issue to resolve.

**Comment by alabulei1** [comment]
@Sush-Dew  Apologies for my delayed response. Yes, this issue is a good starter for you to learn more about WasmEdge. Feel free to reach out me if you have any questions.

**Comment by Its-Maniaco** [comment]
Hello @alabulei1 If I understand this correctly, we are supposed to add unit tests for the lines highlighted in the report?

**Comment by alabulei1** [comment]
> Hello @alabulei1 If I understand this correctly, we are supposed to add unit tests for the lines highlighted in the report?

Yes! It's correct. Do you want to do that? If so, I could assign this issue to you.

**Comment by Its-Maniaco** [comment]
That would be great! Also could you provide me with a little direction. I've written only a few unit tests.

**Comment by hydai** [comment]
Hi @Its-Maniaco 

To increase the code coverage, some new tests need to add to cover this part. For examples, every case should be covered.

**Comment by Its-Maniaco** [comment]
Hello @hydai @alabulei1 I had a good look at the repo. I looked through the test file located in `/WasmEdge/test/host/wasi/wasi.cpp`
I looked through the different TEST functions written to look for some common patterns.
The best I could understand was to create switch statements like we did for `TEST(WasiTest, PollOneoffSocket)` and `TEST(WasiTest, EpollOneoffSocket)` while using values for `__wasi_signal_t` function in `WasmEdge/thirdparty/wasi/api.hpp` file 
but I still cannot figure it out completely.
